### ISSUE TITLE
bug: sync stops in subdirectory after it is moved

### ISSUE THREAD

**Issue by vibs29** [syncthing/syncthing#5173]
<pre>
What happened:
A file added to local host did not get transferred to remote peer.
Syncthing should have transferred it.

Steps to reproduce the problem:

# the sync period is set to 60 seconds
# sync is the directory that is synchronised with remote peer
cd sync
# increment this test directory number every time you run this test
mkdir test6 && cd test6 &&
mkdir -p x/y
# sync, i.e. wait until x/y appears on peer
# then continue as follows on local host
mv x/y 2
mv x 1
sleep 63
touch 2/file1.txt
sleep 63
(peer doesn't get 2/file1.txt)

Version information:

local host
$ syncthing -version
syncthing v0.14.49 "Dysprosium Dragonfly" (go1.10.3 linux-amd64) deb@build.syncthing.net 2018-07-10 15:40:06 UTC [noupgrade]
~$ uname -a
Linux x200 4.9.0-8-amd64 #1 SMP Debian 4.9.110-3+deb9u4 (2018-08-21) x86_64 GNU/Linux

remote peer
$ syncthing -version
syncthing v0.14.39 "Dysprosium Dragonfly" (go1.9 linux-amd64) deb@build.syncthing.net 2017-09-25 06:05:21 UTC [noupgrade]
$ uname -a
Linux workstation 3.16.0-4-amd64 #1 SMP Debian 3.16.39-1+deb8u2 (2017-03-07) x86_64 GNU/Linux
</pre>

**Comment by AudriusButkevicius** [comment]
It's not obvious which machine which commands are supposed to be run on.

**Comment by imsodin** [comment]
From what I understand that's all done on one device. I tried to reproduce, but everything worked just fine here. So either we need a more detailed reproducer (maybe the crucial point is missing) or we need logs that show a problem. Until that's figured out, I propose to move this to the forum.

**Comment by vibs29** [comment]
That's right. It's all done on the local host. This line needs doing manually, and I think is important:
<pre>
# sync, i.e. wait until x/y appears on peer
</pre>

I've tried this experiment 3-4 times. One time, file1.txt was transferred as it should have been, but the remaining times it wasn't.

Note that the two hosts above run different versions of syncthing. Did you use that pair of versions, and both on the versions of Linux shown? If not, maybe the problem is particular to the Syncthing and Linux versions I'm using.

Lastly, I'm happy to submit logs, if you can let me know where to find them and confirm that they don't contain private information.

**Comment by vibs29** [comment]
I've found the cause. It's my fault. This issue is not a syncthing bug.

I decided to check again after one hour, and sure enough file1.txt appeared on the remote peer. Here's what's going on.

The remote peer has "Rescan Interval (s)" set to 60.
It then shared the folder with the local host.
However, on the local host the "Full Rescan Interval (s)" is 3600.

The local host runs a newer Syncthing. I:
- thought the interval was attached to a folder not to a host+folder
- didn't notice the interval field on the local host because the UI has changed

So I ended up leaving it at the default of 3600 on the local host.

In addition, the local host was propagating most changes within a minute. So it wasn't (isn't) apparent to me when propagations were immediate, and when they would only take place during a scan. I take it a scan is for whatever the immediate propagation misses. And this was a case that the immediate propagation missed.

I'm going to go through all my synced folder configurations on all my computers and correct the rescan interval for each host+folder pair, now that I understand Syncthing better.
